temporal-sdk/src/main/java/io/temporal/workflow/Workflow.java

```diff
@@ -615,6 +615,14 @@ public static WorkflowInfo getInfo() {
     return WorkflowInternal.getWorkflowInfo();
   }
 
+  public static <T> Object getMemo(String key, Class<T> valueClass) {
+    return getMemo(key, valueClass, valueClass.getGenericSuperclass());
```

Comment:
valueClass.getGenericSuperclass() doesn't make too much sense. It works because it effectively is not used if valueClass doesn't have generics. Just pass valueClass here for simplicity.